### ISSUE TITLE
Allow `capture_exception` to manually bypass configured `excluded_exceptions`

### ISSUE THREAD

**Issue by ty-porter** [getsentry/sentry-ruby#2006]
## **Describe the idea**
Currently, sentry-ruby filters common controller exceptions out of the box via an array of default `excluded_exceptions` in the configuration. Most of the time, this does a great job cutting down noisy exceptions caused by common user errors like 404s.

However, there's no way to avoid filtering exceptions, and that means they are ALWAYS dropped, even when they are potentially useful information (specifically `ActiveRecord::RecordNotFound`, which I know comes from sentry-rails but the implementation for exclusion occurs in sentry-ruby).

Allow users to manually override the filter, perhaps as such:

```ruby
Sentry.capture_exception(exception, ignore_excluded: true)
```

## **Why do you think it's beneficial to most of the users**
As far as I can tell, other Sentry SDKS for other languages and frameworks do not implement exclusions as aggressively.

Sometimes, it's possible for the default exclusions to contain useful errors, perhaps in a complicated service where the failure might indicate a problem with application code and not necessarily user error.

```ruby
# Will throw ActiveRecord::RecordNotFound
# but this indicates a problem with application code NOT user error (404)!
def always_fails
  Thing.find(-1)
rescue StandardError => exception
  Sentry.capture_exception(exception)
end
```

## **Possible implementation**
#2007

**Comment by st0012** [comment]
I think this is a reasonable request. @sl0thentr0py do other SDKs have mechanisms for this?

**Comment by sl0thentr0py** [comment]
there's no official exposed argument in any other SDK and I wouldn't touch the function signature but we can just hack it into `hint` like we do with `background`? The use case is valid.

**Comment by ty-porter** [comment]
For more context my PR is based around similar changes from #1550

I don't have strong feelings either way, I don't personally use the `hint` arg so if it makes more sense there I'm happy to change

**Comment by st0012** [comment]
I think using `hint` is a good idea. @ty-porter would you mind changing your implementation to use it? Thx

**Comment by st0012** [comment]
Implemented in #2007
